### ISSUE TITLE
OOMKills after upgrading to 0.9.11

### ISSUE THREAD

**Issue by muffl0n** [external-secrets/external-secrets#3145]
**Describe the bug**
With `0.9.11` we're experiencing frequent OOMKills. After downgrading back to `0.9.10` the OOMKills stop.

**To Reproduce**
1. Install version `0.9.11`

**Expected behavior**
No OOMKills

**Screenshots**
![screenshot 2024-02-07 um 09 41 08](https://github.com/external-secrets/external-secrets/assets/927904/622a820e-ffde-46b5-930c-72ae75cb4ed2)

**Additional context**
* ESO: `0.9.11`
* Secret provider: `gcpsm`
* Kubernetes: `v1.27.10-gke.1152000`
* Example of a resource `ExternalSecret` we're using 
```
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  annotations:
    meta.helm.sh/release-name: foo-bar
    meta.helm.sh/release-namespace: foo
  labels:
    app.kubernetes.io/managed-by: Helm
  name: saxon-license
  namespace: foo
spec:
  data:
  - remoteRef:
      conversionStrategy: Default
      decodingStrategy: None
      key: saxon-license
      metadataPolicy: None
      version: latest
    secretKey: saxon-license.lic
  refreshInterval: 1m0s
  secretStoreRef:
    kind: ClusterSecretStore
    name: gcpsm
  target:
    creationPolicy: Owner
    deletionPolicy: Retain
    name: saxon-license
```

**Comment by gusfcarvalho** [comment]
hi @muffl0n ! Can you confirm by setting `refreshInterval` to 10m makes the OOM happen after more time?

**Comment by gusfcarvalho** [comment]
I believe this might be related to the templating feature, which means every external-secrets deployment (regardless of provider) is with memory leaks. @Skarlso @knelasevero

**Comment by Skarlso** [comment]
But there is no tempting in the attached sample. 🤔

**Comment by muffl0n** [comment]
I think we can close this issue. I tried the versions `0.9.10`, `0.9.11` and `0.9.12` once again and this is the result:
![eso](https://github.com/external-secrets/external-secrets/assets/927904/21251c41-83b2-432c-8022-43a030d5ae52)

As written in Slack, I couldn't install `0.9.12` the day I was experiencing the OOMKills cause the Helm chart was not available. Seems like the memory leak was fixed with that release.
